### ISSUE TITLE
mountPoints should default to readOnly=true

### ISSUE THREAD

**Issue by globalcitizen** [appc/spec#391]
A poor selection of default if following the 'deny all, permit explicitly' mantra which is well-respected in the security world.

**Comment by thockin** [comment]
While I appreciate the security world very much, the vast majority of
volumes will be mounted read-write, so making that the default makes sense
to me.  Also, since volumes model mount(), the defaults should be similar -
mount() is read-write unless otherwise instructed.

**Comment by globalcitizen** [comment]
If you're arguing for optimizing a complex system based upon convenience instead of well known security processes, history shows that you're really asking for false economy and problems down the line. Sure, they might not be *your* problems, but someone will have them. It'd be best to change sooner rather than later.

**Comment by runningwild** [comment]
Currently because ReadOnly is a *bool (in types.Volume) if you try to json.Unmarshal something that doesn't specify the ReadOnly you get some comical nonsense like this (note the PANIC inside of it):

```
{0.7.1 PodManifest [{rocketpack-io-flow-stora {<nil> sha512-06959624b668ead2cde5f2de128eee2ff9c0a4c9b84131604afa58838fa92d1e []} 0xc8200bc0e0 [] [{rkt.kubernetes.io/containerhash 2326094207}]}] [%!v(PANIC=runtime error: invalid memory address or nil pointer dereference) default-token-spuha,kind=host,readOnly=falsesource=/var/lib/kubelet/pods/6cad6a24-a27d-11e5-ae42-42010af00003/volumes/kubernetes.io~secret/default-token-spuha] [] [{rkt.kubernetes.io/managed-by-kubelet true} {rkt.kubernetes.io/uid 6cad6a24-a27d-11e5-ae42-42010af00003} {rkt.kubernetes.io/name rocketpack-io-flow-stora-yru1p} {rkt.kubernetes.io/namespace default} {rkt.kubernetes.io/created 1450109544} {rkt.kubernetes.io/restartcount 0}] []}
```

which leads to some confusing bugs.  There hasn't been any discussion on this in quite a while, but should we pick a default so we can depointerify it?  Alternatively I can just manually check this value when it's used (in the String() method), but I figured if I was going to muck around with this we could just give it a default.

btw I do not care what the default is, I just want my json to parse properly.

**Comment by jonboulle** [comment]
@runningwild I am pretty sure that was fixed via https://github.com/appc/spec/pull/541 - mind giving 0.7.2 a try?

**Comment by runningwild** [comment]
Yes you're right, I hadn't updated in a while I guess, my mistake.
